### PR TITLE
Composite formatting: fixed processing order description

### PR DESCRIPTION
**Pull request by pkulikov**
The step 3c of the [composite formatting processing order](https://docs.microsoft.com/en-us/dotnet/standard/base-types/composite-formatting#processing-order) says:
>For objects of other types, if a composite formatting is called with an `IFormatProvider` argument, its value (including a `null`, if no `IFormatProvider` object is supplied) is passed directly to the `IFormattable.ToString` implementation. Otherwise, a `CultureInfo` object that represents the current thread culture is passed to the `IFormattable.ToString` implementation.

The second sentence is redundant. The first sentence covers two complimentary cases: when an `IFormatProvider` argument is supplied and not supplied. I've moved no-`IFormatProvider` argument case into the corrected second sentence.

I've used the following code to check the actual behavior (if you want to run it, just copy it to https://try.dot.net/):
```csharp
using System;
using System.Globalization;

public class Program
{
  public static void Main()
  {
    CultureInfo.CurrentCulture = CultureInfo.GetCultureInfo("en-EN");
    var ghost = new Ghost();
    Console.WriteLine($"I'm a ghost: {ghost:ghost}");
  }

  public class Ghost : IFormattable
  {
    public string ToString(string format, IFormatProvider formatProvider)
    {
      if (format == null) return null;
      if (formatProvider == null) return "not cultural ghost";
      return format.Equals("ghost") ? "~~~" : null;
    }
  }
}
```

Output:
```
I'm a ghost: not cultural ghost
```

[The source code](https://github.com/dotnet/coreclr/blob/2250e2ae2d733934b3aba0a18706a546f9c5fac3/src/mscorlib/shared/System/Text/StringBuilder.cs#L1605) proves that `provider` is just passed to `IFormattable.ToString` method (it's `null` if it is not given to the calling `string.Format` method).

### CODE REVIEW

File: docs/standard/base-types/composite-formatting.md [hunk 1]
```diff
@@ -132,7 +132,7 @@ The .NET composite formatting feature takes a list of objects and a composite fo
...
   
-    -   For objects of other types, if a composite formatting is called with an <xref:System.IFormatProvider> argument, its value (including a `null`, if no <xref:System.IFormatProvider> object is supplied) is passed directly to the <xref:System.IFormattable.ToString%2A?displayProperty=nameWithType> implementation.  Otherwise, a <xref:System.Globalization.CultureInfo> object that represents the current thread culture is passed to the <xref:System.IFormattable.ToString%2A?displayProperty=nameWithType> implementation.  
+    -   For objects of other types, if a composite formatting is called with an <xref:System.IFormatProvider> argument, its value is passed directly to the <xref:System.IFormattable.ToString%2A?displayProperty=nameWithType> implementation. Otherwise, `null` is passed to the <xref:System.IFormattable.ToString%2A?displayProperty=nameWithType> implementation.
```
Comment (rpetrusha) [inline docs/standard/base-types/composite-formatting.md]: unrelated to your PR: It seems that "method" has at some point dropped out of the text -- this was intended to read "if a composite formatting method is called".

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thanks, @pkulikov, for the clarification and correction. There's one issue unrelated to your PR that I'm hoping you can address -- the word "method" has at some point dropped out of the text.

**Comment by pkulikov** [comment]
@rpetrusha thank you for review and noticing a missing word. I've brought it back.

**Review by rpetrusha — APPROVED** [review]
Thanks, @pkulikov, for making the additional correction. I'll merge your PR now.

**Pull request merged**